src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -87,7 +87,7 @@ protected virtual int GetLogicalPinNumber(int pinNumber)
     /// The driver attempts to open the pin without changing its mode or value.
     /// </summary>
     /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
-    public void OpenPin(int pinNumber)
+    public GpioPin OpenPin(int pinNumber)
```

Comment:
Unfortunately, I think this is a binary breaking change (because the caller must insert an extra `pop` if he doesn't need the return value).

Reply:
I second that. Either we kee this change for v3.0 or we can add a different method

Reply:
Agreed, this can't be done. Even for 3.0, it would be really hard to accept this kind of breaking change. We would need a new API for this.

Reply:
OK, I'll create an OpenPinEx function then. Thanks for the feedbacks

Reply:
I'd be ok with making this tiny breaking change for 3.0, especially given this is source compatible